src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp

```diff
@@ -314,7 +330,15 @@ struct layout {
     }
 
     friend bool operator==(const layout& lhs, const layout& rhs) {
-        return lhs.data_type == rhs.data_type && lhs.format == rhs.format && lhs.size == rhs.size && lhs.data_padding == rhs.data_padding;
+        auto get_pshape = [&](const layout& l){
+            if (l.format != cldnn::format::any && l.size.size() < l.format.dimension()) {
+                auto dims = l.get_dims();
```

Comment:
Note: In the future when we are going to infer with PartialShape in cldnn, I think we need to compare with original PartialShape shape not the converted shape to legacy cldnn tensor dims.